File: app/src/main/java/starbright/com/projectegg/MyApp.java
package starbright.com.projectegg;

import android.app.Application;
import android.os.Build;

import starbright.com.projectegg.dagger.component.AppComponent;
import starbright.com.projectegg.dagger.component.DaggerAppComponent;
import starbright.com.projectegg.dagger.module.AppModule;
import starbright.com.projectegg.dagger.module.DataModule;

/**
 * Created by Andreas on 4/8/2018.
 */

public class MyApp extends Application {

    private static AppComponent mAppComponent;

    @Override
    public void onCreate() {
        super.onCreate();
        mAppComponent = DaggerAppComponent.builder()
                .appModule(new AppModule(this))
                .dataModule(new DataModule(BuildConfig.BASE_URL))
                .build();
    }

    public static AppComponent getAppComponent() {
        return mAppComponent;
    }

    public static boolean isPreLolipop() {
        return Build.VERSION.SDK_INT < Build.VERSION_CODES.LOLLIPOP;
    }
}

File: app/src/main/java/starbright/com/projectegg/MyApp.kt
/*
 * Copyright (c) by Andreas (oentoro.andreas@gmail.com)
 * created at 25 - 7 - 2020.
 */

/**
 * Created by Andreas on 29/9/2018.
 */

package starbright.com.projectegg

import android.app.Application
import android.os.Build
import starbright.com.projectegg.dagger.component.ApplicationComponent
import starbright.com.projectegg.dagger.component.DaggerApplicationComponent
import starbright.com.projectegg.dagger.module.AppModule
import timber.log.Timber

class MyApp : Application() {

    lateinit var appComponent: ApplicationComponent

    override fun onCreate() {
        super.onCreate()
        appComponent = DaggerApplicationComponent.builder()
            .appModule(AppModule(this))
            .build()

        if(BuildConfig.DEBUG){
            Timber.plant(Timber.DebugTree())
        }
    }

    companion object {
        val isPreLolipop: Boolean
            get() = Build.VERSION.SDK_INT < Build.VERSION_CODES.LOLLIPOP
    }
}
